### PR TITLE
Treat type equivalent to Type[Any]

### LINKED ISSUE #2655: Type[Any] and type are not equivalent
From `typing` docs:

> `Type[Any]` is equivalent to `Type` which in turn is equivalent to `type`, which is the root of Python’s metaclass hierarchy.

But this code produces error:
```python
from typing import *
y: type
x: Type[Any] = y
```

> error: Incompatible types in assignment (expression has type "`type`", variable has type `Type[Any])`

### PR DESCRIPTION
**Pull request by ilevkivskyi**
Fixes #2655 

As discussed in typing docs, ``Type[Any]`` is equivalent to plain ``type``. ``is_subtype`` was checking this only one side, here I add the opposite side.

### CODE REVIEW

**Comment by ddfisher** [comment]
Instead of fixing this up in one particular place, I think it'd be better to actually translate all `builtins.type` types to `Type[Any]`.

**Comment by elazarg** [comment]
I would expect `type` to be compatible with `Type[T]` for any T. Is this the case in this PR?

**Comment by ilevkivskyi** [comment]
@ddfisher I don't think it is a good idea to translate ``type`` _everywhere_. For example, ``Type`` is not valid as a base class (for a good reason, at runtime it is just a normal class, not a metaclass, and inheriting from ``Type`` leads to unexpected behaviour), it is not valid in runtime context, etc. In addition this could complicate #999 and #2428.

However, I agree that it makes sense to translate ``type`` everywhere in ``subtypes``. I fixed this in a new commit and this also allowed to remove few other special cases. (Note that this means that ``type`` will be subtype of all ``Type[t]``, since ``Type`` is considered covariant and ``Any`` is subtype of everything)

@elazarg Currently all ``Type[t]`` already considered subtypes of ``type``, I only fix here the opposite direction.

File: mypy/test/testsubtypes.py [hunk 1]
```diff
@@ -168,7 +168,7 @@ def test_var_arg_callable_subtyping_9(self) -> None:
             self.fx.callable_var_arg(0, self.fx.b, self.fx.d))
 
     def test_type_callable_subtyping(self) -> None:
-        self.assert_strict_subtype(
+        self.assert_subtype(
```
Comment (ddfisher) [inline mypy/test/testsubtypes.py]: Why this change?  It doesn't seem right to me.

Comment (ddfisher) [inline mypy/test/testsubtypes.py]: Ah, nevermind -- I see the preexisting comment about it in `visit_type_type`.

**Comment by ddfisher** [comment]
That a good point -- I guess we can't translate `type` to `Type[Any]` everywhere.  I'm not sure this is sufficient, though -- will `type` and `Type` interact properly with `meet`, `join`, and TypeVars?

**Comment by elazarg** [comment]
I am not sure why we can't - we can always special case the places where Python happen "not to work the way it ought to" (we won't forget this information).

**Comment by ilevkivskyi** [comment]
@ddfisher Surprisingly (or rather not) it is already treated correctly everywhere else where I found. For example, ``meet_type(Type[whatever], type) == Type[whatever]``, and ``joint_types(Type[whatever]. type) == type``. I added more tests to ensure this in a new commit.

**Comment by ilevkivskyi** [comment]
@ddfisher Are you happy with this PR? Or should I make more changes? (If you don't have time now, just ignore this ping)

**Comment by ilevkivskyi** [comment]
@ddfisher Sorry for pinging again. Are you happy with this PR now?
